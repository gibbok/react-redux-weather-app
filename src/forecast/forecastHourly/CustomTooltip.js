import React from 'react'

const CustomTooltip = React.createClass({
  render () {
    const { payload, active } = this.props
    if (!active) {
      return null
    }
    const data = payload[0].payload
    const hour = data.hour
    const temp = data.temp
    const rain = data.rain ? payload[0].payload.rain.toFixed(1) : 'N/A'
    const day = data.day
    return (
      <div>
        <div>
          {day}
        </div>
        <div>
          <div>
              Hour:
            </div>
          <div>
            {hour}
          </div>
        </div>
        <div>
          <div>
            Temp:
          </div>
          <div>
            {temp} &#176;
            </div>
        </div>
        <div>
          <div>
            Rain:
          </div>
          <div>
            {rain}
          </div>
        </div>
      </div>
    )
  }
})

export default CustomTooltip
