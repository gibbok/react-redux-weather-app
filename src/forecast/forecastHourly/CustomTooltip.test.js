import React from 'react'
import {mount} from 'enzyme'
import toJson from 'enzyme-to-json'
import CustomTooltip from './CustomTooltip'

describe('<CustomTooltip />', () => {
  it('should mount in a full DOM with rain information', () => {
    const wrapper = mount(
      <CustomTooltip
        payload={
        [
          {
            momentId: '2017:8:14:4',
            timespan: 1505401200,
            temp: 23,
            weatherIconCode: 500,
            rain: 0.15,
            day: 'Thu, Sep 14',
            hour: '05 PM'
          }
        ]}
    />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should mount in a full DOM without rain information', () => {
    const wrapper = mount(
      <CustomTooltip
        payload={
        [
          {
            momentId: '2017:8:14:4',
            timespan: 1505401200,
            temp: 23,
            weatherIconCode: 500,
            rain: 'N/A',
            day: 'Thu, Sep 14',
            hour: '05 PM'
          }
        ]}
    />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
