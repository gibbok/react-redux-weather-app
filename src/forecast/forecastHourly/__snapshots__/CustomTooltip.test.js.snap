// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomTooltip /> should mount in a full DOM with rain information 1`] = `
<CustomTooltip
  payload={
    Array [
      Object {
        "day": "Thu, Sep 14",
        "hour": "05 PM",
        "momentId": "2017:8:14:4",
        "rain": 0.15,
        "temp": 23,
        "timespan": 1505401200,
        "weatherIconCode": 500,
      },
    ]
  }
/>
`;

exports[`<CustomTooltip /> should mount in a full DOM without rain information 1`] = `
<CustomTooltip
  payload={
    Array [
      Object {
        "day": "Thu, Sep 14",
        "hour": "05 PM",
        "momentId": "2017:8:14:4",
        "rain": "N/A",
        "temp": 23,
        "timespan": 1505401200,
        "weatherIconCode": 500,
      },
    ]
  }
/>
`;
